.github/workflows/ci.yml

```diff
@@ -0,0 +1,73 @@
+name: semmc
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  build-linux:
+    strategy:
+      fail-fast: false
+      matrix:
+        ghc: ["8.6.5", "8.8.4", "8.10.2"]
+        package:
+          - semmc
+          - semmc-aarch32
+          - semmc-arm
+          - semmc-ppc
+          - semmc-x86_64
+          - semmc-coverage
+          - semmc-fuzzer
+          - semmc-toy
+          - semmc-synthesis
+          - semmc-learning
+    runs-on: ubuntu-latest
+    name: ${{ matrix.package }} - GHC v${{ matrix.ghc }} - ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          git config --global url.https://github.com/.insteadOf git@github.com:
+          git submodule update --init
+
+      - uses: actions/setup-haskell@v1
+        id: setup-haskell
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          enable-stack: true
+      - uses: actions/cache@v2
+        name: Cache cabal store
+        with:
+          path: |
+            ${{ steps.setup-haskell.outputs.cabal-store }}
+            dist-newstyle
+          key: cabal-${{ matrix.package }}-${{ matrix.ghc }}-${{ hashFiles('**/cabal.GHC-*') }}-${{ github.sha }}
+          restore-keys: |
+            cabal-${{ matrix.package }}-${{ matrix.ghc }}-${{ hashFiles('**/cabal.GHC-*') }}
+
+      - name: Configure
+        run: |
+          ln -sv cabal.project.newbuild cabal.project
+          ln -sv cabal.GHC-${{ matrix.ghc }}.config cabal.project.freeze
+          cabal v2-update
+          cabal v2-configure -j2 --minimize-conflict-set
```

Comment:
Probably want to add `--enable-tests` to this line to ensure tests are built.

Reply:
Looks like tests for semmc have some issues :(. I've selectively disabled them in this PR so they can be handled separately.

Reply:
https://github.com/GaloisInc/semmc/pull/57 was opened for handling the failing semmc tests. This PR should be good to go in if you're happy with what's enabled here so far.